src/main/java/cz/cvut/kbss/termit/security/JwtAuthorizationFilter.java

```diff
@@ -76,17 +79,49 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
             SecurityUtils.verifyAccountStatus(existingDetails.getUser());
             securityUtils.setCurrentUser(existingDetails);
             refreshToken(authToken, response);
-        } catch (DisabledException | LockedException | JwtException | UsernameNotFoundException e) {
-            response.setStatus(HttpStatus.UNAUTHORIZED.value());
-            objectMapper.writeValue(response.getOutputStream(),
-                    ErrorInfo.createWithMessage(e.getMessage(), request.getRequestURI()));
+        } catch (JwtException e) {
+            if (shouldAllowThroughUnauthenticated(request)) {
+                chain.doFilter(request, response);
+                return;
+            } else {
+                unauthorizedRequest(request, response, e);
+                return;
+            }
+        } catch (DisabledException | LockedException | UsernameNotFoundException e) {
+            unauthorizedRequest(request, response, e);
             return;
         }
         chain.doFilter(request, response);
```

Comment:
why not to put chain.doFilter into the try block and avoid the return statements at all?